include/caf/events/power_event.h

```diff
@@ -19,13 +20,59 @@
 extern "C" {
 #endif
 
+/** @brief Power down event.
+ *
+ * The module that controls power management in the application can submit the power down event to
+ * inform other application modules that the modules should be suspended to reduce the power
+ * consumption. The power down event can also be submitted to suspend application modules if an
+ * application module reports fatal error. An application module can report fatal error submitting
+ * @ref module_state_event with state set to MODULE_STATE_ERROR.
+ *
+ * An application module that handles the power down event should submit @ref module_state_event
+ * with state set to MODULE_STATE_OFF or MODULE_STATE_STANDBY. The MODULE_STATE_STANDBY should
+ * be submitted by a module that is capable of submitting @ref wake_up_event to wake up the system
+ * on external event. Otherwise MODULE_STATE_OFF should be reported. If the module is unable to
+ * suspend at the moment, it should consume the received power down events until it's ready to
+ * suspend itself. The module should also delay submitting the @ref module_state_event until it's
+ * ready to suspend itself.
+ *
+ * The module may handle the power down event differently if it's related to fatal error.
+ * In general, an application module should suspend itself, but some of the modules may still need
+ * to be active after the fatal error. For example, module that controls hardware LEDs may be used
+ * to display LED effect related to the fatal error.
+ *
+ * The module that controls power management is the final subscriber for the power down event.
+ * When this module receives the power down event, it's ensured that all other application modules
+ * that handle power down event are already suspended. Then power down procedure can be continued.
+ *
+ * @warning An application module that handles power down event must also handle @ref wake_up_event.
+ *          Otherwise the module will never be woken up after suspending.
+ */
```

Comment:
Discussed offline:
I think we should not describe in detail what power management module should look like and how modules should submit/react to `wake_up/power_down _event`. Here we should only mention what the events are. All additional information would be put to other place.

This part of documentation should be shortened to something like:
The power down event is used to inform application modules that each of them should enter suspended mode, where power consumption is reduced and for modules to signalize fatal error.

Reply:
Added this at the beginning of the event's documentation